builtin/providers/aws/resource_aws_cloudwatch_metric_alarm_test.go

```diff
@@ -0,0 +1,98 @@
+package aws
+
+import(
+    "fmt"
+//    "math/rand"
+//    "strings"
+    "testing"
+//    "time"
+//
+    "github.com/awslabs/aws-sdk-go/aws"
+    "github.com/awslabs/aws-sdk-go/service/cloudwatch"
+    "github.com/hashicorp/terraform/helper/resource"
+    "github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccAWSCloudWatchMetricAlarm_basic(t* testing.T) {
+    var alarm cloudwatch.MetricAlarm
+
+    resource.Test(t, resource.TestCase{
+        PreCheck:       func () { testAccPreCheck(t) },
+        Providers:      testAccProviders,
+        CheckDestroy:   testAccCheckAWSCloudWatchMetricAlarmDestroy,
+        Steps:          []resource.TestStep{
+            resource.TestStep{
+                Config: testAccAWSCloudWatchMetricAlarmConfig,
+                Check:  resource.ComposeTestCheckFunc(
+                    testAccCheckCloudWatchMetricAlarmExists("aws_cloudwatch_metric_alarm.foobar", &alarm),
```

Comment:
I have `testAccCheckCloudWatchMetricAlarmExists` that checks the existence of the alarm object. Is this perhaps unnecessary? I modeled this after some of the other (what look to be older styles) test cases.